### ISSUE TITLE
MelDataset mel VS mel_loss

### ISSUE THREAD

**Issue by nikifori** [jik876/hifi-gan#163]
What's the difference between those two mel spectrograms that MelDataset returns (mel and mel_loss)?
What's the difference of the former alleged mels from Tacotron 2?

Thanks

**Comment by msalhab96** [comment]
You can think of `mel_loss` as the ground truth mel-spectrogram, and `mel` is the input mel-spectrogram.

So in case you are doing pertaining, they both will be the same, if you are doing fine-tuning, then `mel` is the input which generated from tacotron, and `mel_loss` is the ground truth.

check the code [here](https://github.com/jik876/hifi-gan/blob/4769534d45265d52a904b850da5a622601885777/meldataset.py#L143)

**Comment by Ziyi6** [comment]
@msalhab96 Hello. I noticed that in training,  we use the LJ-Speech dataset which contains many wav files. What we do in training is that we extract the mel spectrogram from these wavs with highest frequency up tp 8kHz (see fmax in config file) and feed them to the model. The model outputs wavs, then  we calculate the mel-spectrogram again, but with highest frequency up to 11.025kHz (see fmax_for_loss=null in config file, in librosa, the default value of highest frequency is half of the sampling rate) for calculating the loss. Do you know why is there a gap?
